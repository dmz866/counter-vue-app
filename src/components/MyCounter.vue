<template>
    <h3 class="bg-red text-red">Counter: {{ counter }}</h3>
    <h1 data-testid="squared-label">Squared Counter: {{ squaredCounter }}</h1>
    <button class="bg-blue-100 text-red-300" @click="add">+1</button>
    <button @click="minus">-1</button>
</template>

<script lang="ts" setup>
import { useCounter } from '../composables/useCounter';

//const { value } = defineProps({ value: { type: Number, required: true } });
const { value } = defineProps<{ value: number }>();
const { counter, squaredCounter } = useCounter(value);
const add = () => counter.value++;
const minus = () => counter.value--;
</script>

<style scoped>
h1 {
    color: aqua;
}
</style>
