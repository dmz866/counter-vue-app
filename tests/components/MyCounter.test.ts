import { mount } from "@vue/test-utils";
import { describe } from "node:test";
import { expect, test } from "vitest";
import MyCounter from '../../src/components/MyCounter.vue';

describe('Test My Counter component', () => {
  test('should match snapshot', () => {
    const wrapper = mount(MyCounter);

    expect(wrapper.html()).toMatchSnapshot();
  });

  test('should render the counter value correctly', () => {
    const value = 5;
    const wrapper = mount(MyCounter, {
      props: { value }
    });

    expect(wrapper.find('h3').text()).toContain(`Counter: ${value}`);
    expect(wrapper.find('[data-testid="squared-label"]').text()).toContain(`Squared Counter: ${value * value}`);
  });

  test('should increment the counter when +1 button is clicked', async () => {
    const value = 5;
    const wrapper = mount(MyCounter, {
      props: { value }
    });
    const btn = wrapper.find('button');
    await btn.trigger('click');

    expect(wrapper.find('h3').text()).toContain(`Counter: ${value + 1}`);
  });
});
