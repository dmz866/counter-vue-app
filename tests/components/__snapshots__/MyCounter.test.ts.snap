// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should match snapshot 1`] = `
"<h3 data-v-0352012b="" class="bg-red text-red">Counter: </h3>
<h1 data-v-0352012b="" data-testid="squared-label">Squared Counter: NaN</h1>
<button data-v-0352012b="" class="bg-blue-100 text-red-300">+1</button>
<button data-v-0352012b="">-1</button>"
`;
